runtime/vm/resolvesupport.cpp

```diff
@@ -863,8 +869,27 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 		nameAndSig = J9ROMFIELDREF_NAMEANDSIGNATURE(romFieldRef);
 		name = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
 		signature = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
-		fieldOffset = instanceFieldOffsetWithSourceClass(vmStruct, resolvedClass, J9UTF8_DATA(name), J9UTF8_LENGTH(name), J9UTF8_DATA(signature), J9UTF8_LENGTH(signature), &definingClass, (UDATA *)&field, lookupOptions, classFromCP);
-		
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		if (J9_IS_J9CLASS_VALUETYPE(resolvedClass)) {
```

Comment:
add a comment saying: This is an optimization that searches for a field offset in the FCC. If the offset is found there is no need to repeat the process. Also, since this optimization is only done for ValueTypes, the resolvedClass will always be the class that owns the field, since ValueType superclasses can not have fields.